### PR TITLE
fix(aws-amplify-react) disables signup button while request is in flight

### PR DESCRIPTION
**Pull request by haverchuck**
_Issue #, if available:_
Fixes: #3558 

_Description of changes:_
 Disables signup button while request is in flight by using a requestPending property in state.  It is implemented at the AuthPiece level, so should be reusable in other Auth components (but making this PR just for signup since the bug is more than just a nuissance but actually results in multiple user pool users).

To test, trying creating a pre-signup lambda trigger with a timeout wrapping the response.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/aws-amplify-react/__tests__/Auth/SignUp-test.tsx [hunk 1]
```diff
@@ -125,6 +125,39 @@ describe('signUp without signUpConfig prop', () => {
 			spyon_changeState.mockClear();
 		});
 
+		test('state.requestPending shoudl be true when signUp execution begins', () => {
```
Comment (Amplifiyer) [inline packages/aws-amplify-react/__tests__/Auth/SignUp-test.tsx]: Nit: typo

Comment (haverchuck) [reply packages/aws-amplify-react/__tests__/Auth/SignUp-test.tsx]: fixed :)

**Comment by haverchuck** [comment]
@sammartinez - I created the tasks and wrote some tests

File: packages/aws-amplify-react/src/Auth/SignUp.tsx [hunk 2]
```diff
@@ -327,6 +333,7 @@ export default class SignUp extends AuthPiece<ISignUpProps, IAuthPieceState> {
 				<SectionFooter theme={theme} data-test={auth.signUp.footerSection}>
 					<SectionFooterPrimaryContent theme={theme}>
 						<Button
+							disabled={this.state.requestPending}
```
Comment (ashika01) [inline packages/aws-amplify-react/src/Auth/SignUp.tsx]: Just a NIT: we could destructure the state like we have done with props :)

**Pull request merged**